syntax = "proto3";

package hello;

service HelloService {
    rpc SayHello (HelloRequest) returns (HelloResponse);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string greeting = 1;
}


service FraudDetectionService {
    rpc DetectFraud(FraudDetectionRequest) returns (FraudDetectionResponse);
}

message FraudDetectionRequest {
    string creditCardNumber = 1;
    string creditCardExpirationDate = 2;
    string creditCardCVV = 3;
    string discountCode = 4;
}

message FraudDetectionResponse {
    bool isFraud = 1;
    optional string message = 2;
}